# How Sleep Affects Health Outcomes

>Name: Brynn O'Toole

## Python Setup

In [5]:
# Import packages for numerical computation, statistics, plotting, and data wrangling
import numpy as np
import scipy.stats as stats
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

## Introduction

Sleep is a fundamental biological process that plays a critical role in maintaining both physical and mental health. Adequate sleep supports essential functions such as immune regulation, metabolic balance, cardiovascular health, and brain performance. Research shows that insufficient sleep is linked to a wide range of negative health outcomes, including increased risk of chronic diseases like heart disease, obesity, and diabetes, as well as impaired cognitive functioning and emotional regulation . In addition to physical effects, sleep is deeply connected to mental health, as it helps regulate mood, stress responses, and the brain’s ability to process emotions. Without enough sleep, the body and mind are less equipped to recover from daily stressors, leading to both short-term and long-term health consequences. [Arambh Sanjay Shah](https://journals.sagepub.com/doi/10.1177/15598276251346752)

A growing body of research highlights a strong relationship between sleep duration and mental health outcomes, particularly stress and anxiety. Studies consistently show that people who experience sleep deprivation report higher levels of emotional distress, including increased anxiety and reduced ability to manage stress effectively. For example, research from the American Psychological Association found that chronic sleep loss can *undermine emotional functioning* and increase the risk of anxiety symptoms [Palmer, C. A.](https://psycnet.apa.org/record/2024-34959-001?doi=1). Similarly, otherwise healthy individuals often experience heightened anxiety and distress after poor sleep, demonstrating that even short-term sleep deprivation can negatively affect emotional stability (). These findings suggest that people who sleep less are indeed more likely to report higher levels of stress and anxiety.

Importantly, the relationship between sleep and mental health is bidirectional, meaning that poor sleep can increase stress and anxiety, while stress and anxiety can also disrupt sleep. Research shows that individuals with insomnia are significantly more likely to experience anxiety disorders, and sleep deprivation reduces the brain’s ability to regulate emotions effectively . At the same time, stress itself can directly impair sleep quality, creating a cycle in which poor sleep and heightened stress reinforce one another . Overall, the evidence clearly indicates that sleep is a key factor in health outcomes, and that insufficient sleep is strongly associated with higher levels of stress and anxiety, making it an important area of focus for improving overall well-being.

The aim of this project is to analyze the relationship between sleep patterns and health outcomes using data-driven methods, identifying key associations between sleep duration, quality, and various physical health measures.

## Background: The Relationship Between Sleep and Health Outcomes

Sleep is a fundamental biological process essential for maintaining physical, cognitive, and emotional health. It plays a critical role in regulating numerous physiological systems, including metabolic, cardiovascular, immune, and neurological functions. In recent decades, sleep has become a major focus in public health research due to increasing rates of sleep deprivation and sleep disorders worldwide. Modern lifestyle factors, such as longer work hours, increased screen exposure, and shift based employment—have contributed to widespread disruptions in both sleep duration and quality [Zimmerman M](https://www.sleephealthjournal.org/article/S2352-7218(23)00293-0/abstract?utm_source=chatgpt.com).

Adequate sleep is typically defined as 7–9 hours per night for adults, though both sleep quantity and quality are important determinants of health [Gao C](10.3389/fmed.2021.813943). Research increasingly shows that deviations from this range—either short sleep (generally <6–7 hours) or long sleep (>8–9 hours)—are associated with a wide range of adverse health outcomes. Large-scale meta-analyses and umbrella reviews have demonstrated that short sleep duration is linked to increased risks of obesity, hypertension, and metabolic disorders, while long sleep duration is associated with elevated risks of stroke and all-cause mortality [Gao C](10.3389/fmed.2021.813943). These findings highlight that sleep has a nonlinear relationship with health outcomes, where both insufficient and excessive sleep may signal or contribute to underlying health issues.

One of the primary pathways through which sleep affects health is through metabolic and cardiovascular regulation. Sleep deprivation has been shown to impair glucose metabolism and increase insulin resistance, contributing to the development of type 2 diabetes [NHLBI](https://www.nhlbi.nih.gov/sites/default/files/media/docs/NHLBI_Constituency_Report_2024_SleepHealth_2024.pdf?utm_source=chatgpt.com). Additionally, chronic insufficient sleep is associated with elevated blood pressure and increased risk of cardiovascular disease. Disruptions in circadian rhythms—such as those experienced by shift workers—further exacerbate these risks by misaligning internal biological clocks with environmental cues, leading to long-term physiological stress and inflammation [Regalla S](https://impactfactor.org/PDF/IJTPR/14/IJTPR%2CVol14%2CIssue9%2CArticle5.pdf?utm_source=chatgpt.com).

Sleep also plays a crucial role in brain health and cognitive functioning. During sleep, the brain undergoes processes essential for memory consolidation, emotional regulation, and the removal of metabolic waste products. Disturbed sleep has been linked to impaired cognitive performance, reduced attention, and increased risk of neurodegenerative diseases such as Alzheimer’s disease and dementia [Gottesman R](https://www.ahajournals.org/doi/pdf/10.1161/STR.0000000000000453?utm_source=chatgpt.com). Emerging research suggests that sleep may facilitate the functioning of the glymphatic system, which clears neurotoxic waste from the brain, providing a biological mechanism linking poor sleep to long-term neurological decline.

Mental health outcomes are strongly associated with sleep quality as well. Insufficient or poor-quality sleep has been consistently linked to increased rates of depression, anxiety, and stress-related disorders. Meta-analytic evidence indicates that improvements in sleep quality can lead to significant reductions in symptoms of depression and anxiety, suggesting a potentially causal relationship between sleep and mental health [Scott A](https://www.sciencedirect.com/science/article/pii/S1087079221001416?via%3Dihub). Prolonged sleep deprivation has been shown to impair emotional regulation and increase vulnerability to psychiatric disorders [Palmer C](https://www.apa.org/pubs/journals/releases/bul-bul0000410.pdf?utm_source=chatgpt.com).

Beyond individual physiological systems, sleep is increasingly recognized as a multidimensional construct encompassing duration, quality, timing, and regularity. Recent research emphasizes that sleep regularity—consistent sleep and wake times—may be as important as total sleep duration in predicting health outcomes, including mortality risk and cardiometabolic health. Irregular sleep patterns can disrupt circadian rhythms and negatively affect processes such as hormone regulation, digestion, and immune function.

From a population health perspective, inadequate sleep represents a significant and growing public health concern. A substantial proportion of adults fail to meet recommended sleep guidelines, particularly among populations exposed to high stress, socioeconomic disadvantage, or irregular work schedules. This widespread sleep deficiency has implications not only for individual health but also for healthcare systems, productivity, and safety outcomes (e.g., increased workplace errors and accidents). As a result, sleep is increasingly viewed as a modifiable risk factor that can be targeted through behavioral interventions, policy changes, and technological innovations.

In the context of health data science, the study of sleep and health outcomes presents both opportunities and challenges. Advances in wearable technology, actigraphy, and large-scale cohort studies (e.g., UK Biobank, NHANES) have enabled the collection of high resolution sleep data, facilitating more precise modeling of sleep patterns and their associations with disease outcomes. Analytical approaches such as longitudinal modeling, machine learning, and causal inference are increasingly used to disentangle complex relationships between sleep and health. However, challenges remain in standardizing sleep measurements and accounting for confounding variables such as lifestyle, comorbidities, and socioeconomic factors.

Overall, the existing body of research demonstrates that sleep is a critical determinant of health across multiple domains. Both insufficient and excessive sleep, as well as poor sleep quality and irregular sleep patterns, are associated with adverse physical, mental, and cognitive outcomes. Understanding these relationships is essential for developing effective interventions and improving population health.

## Methods

## Results

### Data Import and Verification

The data are contained in a comma-separated vales file, which is imported below and assigned to the varaible `df`. - need the `` for code stuff.

In [11]:
df = pd.read_csv('/Users/brynnotoole/Documents/pubh1142/Sleep_health_and_lifestyle_dataset.csv')

We perform data verification on the data:
1. Shape
2. Information
3. Return first five rows of data

In [12]:
df.shape

(374, 13)

The dataset contains information about $374$ subjects. The dataset consists of $13$ variables.

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 374 entries, 0 to 373
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Person ID                374 non-null    int64  
 1   Gender                   374 non-null    object 
 2   Age                      374 non-null    int64  
 3   Occupation               374 non-null    object 
 4   Sleep Duration           374 non-null    float64
 5   Quality of Sleep         374 non-null    int64  
 6   Physical Activity Level  374 non-null    int64  
 7   Stress Level             374 non-null    int64  
 8   BMI Category             374 non-null    object 
 9   Blood Pressure           374 non-null    object 
 10  Heart Rate               374 non-null    int64  
 11  Daily Steps              374 non-null    int64  
 12  Sleep Disorder           155 non-null    object 
dtypes: float64(1), int64(7), object(5)
memory usage: 38.1+ KB


In [14]:
df.head()

,Person ID,Gender,Age,Occupation,Sleep Duration,Quality of Sleep,Physical Activity Level,Stress Level,BMI Category,Blood Pressure,Heart Rate,Daily Steps,Sleep Disorder
0,1,Male,27,Software Engineer,6.1,6,42,6,Overweight,126/83,77,4200,NaN
1,2,Male,28,Doctor,6.2,6,60,8,Normal,125/80,75,10000,NaN
2,3,Male,28,Doctor,6.2,6,60,8,Normal,125/80,75,10000,NaN
3,4,Male,28,Sales Representative,5.9,4,30,8,Obese,140/90,85,3000,Sleep Apnea
4,5,Male,28,Sales Representative,5.9,4,30,8,Obese,140/90,85,3000,Sleep Apnea


### Exploratory Data Analysis

The `Sleep Disorder` column contains information contains informations about the outcome of this study. Continue with summary statistics ...... 

Levels of measurement:
1. Numerical: ratio & interval
2. Categorical: numerical & ordinal

#### Sleep Disorder

In [17]:
df['Sleep Disorder']

0              NaN
1              NaN
2              NaN
3      Sleep Apnea
4      Sleep Apnea
          ...     
369    Sleep Apnea
370    Sleep Apnea
371    Sleep Apnea
372    Sleep Apnea
373    Sleep Apnea
Name: Sleep Disorder, Length: 374, dtype: object

Need to get rid of missing values 

Categorical variable 

List all the categories 

In [20]:
df['Sleep Disorder'].unique()

array([nan, 'Sleep Apnea', 'Insomnia'], dtype=object)

In [22]:
# Calculate the frequency of each value in the 'Sleep Disorder' column
df['Sleep Disorder'].value_counts()

Sleep Disorder
Sleep Apnea    78
Insomnia       77
Name: count, dtype: int64

Do this for all the categorical columns 

In this dataset, 78 people have `Sleep Apnea` and 77 people have `Insomnia`.

In [23]:
# Calculate the percentage of each value in the 'Sleep Disorder' column
df['Sleep Disorder'].value_counts(normalize=True) * 100

Sleep Disorder
Sleep Apnea    50.322581
Insomnia       49.677419
Name: proportion, dtype: float64

In this dataset, 50.3% of people have `Sleep Apnea` and 49.7% of people have `Insomnia`. figure out how to do the percentages 

#### Age

In [24]:
# summary statistics for the `Age` column
df['Age'].describe()

count    374.000000
mean      42.184492
std        8.673133
min       27.000000
25%       35.250000
50%       43.000000
75%       50.000000
max       59.000000
Name: Age, dtype: float64

In [25]:
# calculate the interquartile range (IQR) for the `Age` column
Q1 = df['Age'].quantile(0.25)
Q3 = df['Age'].quantile(0.75)
IQR = Q3 - Q1
print(f'IQR for Age: {IQR}')    

IQR for Age: 14.75


In [26]:
stats.iqr(df['Age'])

np.float64(14.75)

Should have range for Age - who is the oldest and youngest etc

In [27]:
# Calculate summary statistics of the `Age` column grouped by the `Sleep Disorder` column
df.groupby('Sleep Disorder')['Age'].describe()

,count,mean,std,min,25%,50%,75%,max
Sleep Disorder,,,,,,,,
Insomnia,77.0,43.519481,4.808464,28.0,43.0,44.0,45.00,53.0
Sleep Apnea,78.0,49.705128,8.990773,28.0,49.0,50.0,57.75,59.0


#### BMI Category

In [31]:
df['BMI Category'].unique()

array(['Overweight', 'Normal', 'Obese', 'Normal Weight'], dtype=object)

In [29]:
# Return unique values in the `BMI Category` column and their frequencies
df['BMI Category'].value_counts() 

BMI Category
Normal           195
Overweight       148
Normal Weight     21
Obese             10
Name: count, dtype: int64

In [32]:
# Calculate the percentages of each value in the `BMI Category` column
df['BMI Category'].value_counts(normalize=True) * 100

BMI Category
Normal           52.139037
Overweight       39.572193
Normal Weight     5.614973
Obese             2.673797
Name: proportion, dtype: float64

Explain numbers 

In [33]:
# Return a contingency table of the `Sleep Disorder` and `BMI Category` columns
pd.crosstab(df['Sleep Disorder'], df['BMI Category'])

BMI Category,Normal,Normal Weight,Obese,Overweight
Sleep Disorder,,,,
Insomnia,7,2,4,64
Sleep Apnea,5,2,6,65
